Sources/GRPC/ConnectionPool/PoolManager.swift

```diff
@@ -353,6 +377,18 @@ internal final class PoolManager {
       self._state.shutdownComplete()
     }
   }
+
+  // MARK: - Stats
+
+  private func emitStats(delegate: GRPCConnectionPoolDelegate) {
+    let pools = self.lock.withLock { self._pools }
+    if pools.isEmpty { return }
+
+    let statsFutures = pools.map { $0.stats() }
```

Comment:
Should we capture the stats while we're holding the lock for the pools? We could for example shut down some of the pools in between the capture and doing this map, which isn't a snapshot of an instant in time of all pools. I don't know if we care about this or not though.

Reply:
I don't think it matters to be honest; each pool needs to exec onto its event loop to get stats so they won't be from the same instant anyway.